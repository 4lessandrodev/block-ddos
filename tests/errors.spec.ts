import { Request } from "express";
import blockDDoS from "../lib";
import { next, request, ResetMock, response, result } from "./utils";

describe('custom errors', () => {
    it('should return success', () => {
        ResetMock();
        const req = Object.assign({}, { ...request }, { socket: { remoteAddress: '60.198.115.122' } }) as Request;
        const middleware = blockDDoS({ error: { result: 'My custom error', isFail: true }, attempts: 1 });
        middleware(req, response, next);
        expect(result).toEqual({ data: { next: true }, status: 200 });
    })

    it('should return custom error', () => {
        ResetMock();
        const req = Object.assign({}, { ...request }, { socket: { remoteAddress: '60.198.115.122' } }) as Request;
        const middleware = blockDDoS({ error: { msg: 'My custom error', isFail: true }, attempts: 1 });
        middleware(req, response, next);
        expect(result).toEqual({ data: { error: { isFail: true, msg: 'My custom error' } }, status: 403 });
    })
});
