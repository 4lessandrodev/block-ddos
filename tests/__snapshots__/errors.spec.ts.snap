// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom errors should block by cookies 1`] = `
Object {
  "cookies": Object {
    "ddos-blocked-times": 1,
  },
  "headers": Object {
    "cookie": "ddos-blocked-times=21",
    "x-forwarded-for": "77.11.107.51",
  },
  "hostname": "http://localhost",
  "method": "GET",
  "path": "/index",
  "protocol": "http",
}
`;

exports[`custom errors should block by secure cookies 1`] = `
Object {
  "cookies": Object {
    "ddos-blocked-times": 1,
  },
  "headers": Object {
    "cookie": "ddos-blocked-times=21",
    "x-forwarded-for": "77.11.107.51",
  },
  "hostname": "http://localhost",
  "method": "GET",
  "path": "/index",
  "protocol": "https",
}
`;
